src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -173,6 +173,33 @@ CORINFO_CLASS_HANDLE Compiler::gtGetStructHandleForHWSIMD(var_types simdType, va
     return NO_CLASS_HANDLE;
 }
 
+#ifdef FEATURE_HW_INTRINSICS
+/* static */ bool Compiler::vnEncodesResultTypeForHWIntrinsic(NamedIntrinsic hwIntrinsicID)
```

Comment:
This needs a function header comment. I don't understand why an intrinsic that takes multiple element types (`baseType`s) implies that the result type needs to be encoded in the VN.